plugins/rzkeychange/rzkeychange.c

```diff
@@ -242,8 +265,31 @@ void rzkeychange_submit_counts(void)
         report_node,
         report_server,
         report_zone);
-    dns_query(qname, LDNS_RR_TYPE_TXT);
-    /* normally we would free any return packet, but this process is about to exit */
+    pkt = dns_query(qname, LDNS_RR_TYPE_TXT);
+    if (pkt)
+        ldns_pkt_free(pkt);
+    if (keytag_zone != 0) {
+        unsigned int i;
+        for (i = 0; i < num_key_tag_signals; i++) {
+            char *s = strdup(rzkeychange_ia_str(key_tag_signals[i].addr));
+            char *t;
+            if (0 == s)
+                break;
+            for (t = s; *t; t++)
+                if (*t == '.' || *t == ':')
+                   *t = '-';
+            snprintf(qname, sizeof(qname), "%lu.%s.%s.%s.%s.%s",
```

Comment:
You should check that it could construct the full string so you don't continue querying broken qnames.